singularity/local.go

```diff
@@ -0,0 +1,163 @@
+package singularity
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/data-preservation-programs/singularity/database"
+	"github.com/data-preservation-programs/singularity/datasource"
+	"github.com/data-preservation-programs/singularity/model"
+	"github.com/data-preservation-programs/singularity/service/datasetworker"
+	fs2 "github.com/rclone/rclone/fs"
+	"gorm.io/gorm"
+)
+
+var (
+	ErrItemNotFound   = errors.New("no item is found with the given id")
+	ErrSourceNotFound = errors.New("no source is found with the given id")
+	ErrNotInitialized = errors.New("singularity source not initialized")
+)
+
+const MOTION_DATASET_NAME = "MOTION_DATASET"
+const PIECE_SIZE = 1 << 36
+const MAX_CAR_SIZE = (1 << 29) * 127
+
+type LocalSingularity struct {
+	db                        *gorm.DB
```

Comment:
It seems unfortunate to have to depend on gorm just to use Singularity. 

In the long run I think Singularity should be refactored into a reusable  library and avoid exposing database models directly to the users. All sorts of funky could happen if it's exposed.